src/main/java/com/google/sps/servlets/SentimentServlet.java

```diff
@@ -52,40 +56,62 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     } catch (GoogleJsonResponseException e) { // some other problem with requesting comment data
       response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, COMMENTS_FAILED_ERROR);
     }
-    String comments = String.join(". ", commentsList);
 
     String captions = captionService.getCaptionFromId(videoId);
 
-    if (comments.isEmpty() && captions.isEmpty()) {
-      VideoAnalysis videoAnalysis = new VideoAnalysis.Builder()
-          .setScore(null)
-          .setScoreAvailable(false)
-          .build();
-      String json = new Gson().toJson(videoAnalysis);
-
+    if (commentsList.isEmpty() && captions.isEmpty()) {
+      String json = createResponseJson(null, false);
       response.setContentType("application/json;");
       response.getWriter().println(json);
       return;
     }
+    
+    ExecutorService executor = Executors.newFixedThreadPool(26);
 
-    String text = comments + ". " + captions;
+    List<Future> commentScoreFutures = new ArrayList<>();
+    for (String comment : commentsList) {
+      Future<Float> future = executor.submit(new Callable<Float>() {
+        @Override
+        public Float call() {
+          return calculateSentimentScore(comment);
+        }
+      });
+      commentScoreFutures.add(future);
+    }
 
-    Float score = null;
+    Future<Float> captionFuture = executor.submit(new Callable<Float>() {
+      @Override
+      public Float call() {
+        return calculateSentimentScore(captions);
+      }
+    });
+    
+    Float commentsScore = 0f;
+    Float captionsScore = null;
```

Comment:
Are they intentionally initialized in different ways?

Reply:
hmm I guess they could both be initialized to 0, but basically I had null for both at first and then realized commentsScore has to be 0 because you're incrementing it as you retrieve the score from each comment. I'll change it so that they're both 0